library/include/ck/library/utility/literals.hpp

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+// [P0330] Literal Suffix for (signed) size_t (C++23)
+// ref: https://wg21.link/p0330r8
+inline constexpr std::size_t operator""_uz(unsigned long long size)
```

Comment:
This is using operator loading to mimic a C++23 feature?

What is this C++ magical feature ```operator""_uz```, is it operator overloading? Do you have some reference for this feature?

Reply:
This is a C++11 feature: [_user-defined literals_](https://en.cppreference.com/w/cpp/language/user_literal)
User define literals must have underscore `_` prefix (to distinguish from the standard ones, e.g. `std::string` literal [`""s`](https://en.cppreference.com/w/cpp/string/basic_string/operator%22%22s))
From C++23, ISO committee have added `z`/`uz ` for (signed) `std::size_t` literals, so I think it is nice to have it too in CK (even we can only use library solution).